### DISCUSSION TITLE
Why are Origin and Edge servers needed, and how does caching work?

### DISCUSSION THREAD

**Discussion by mauricioabreu** [OvenMediaLabs/OvenMediaEngine#1674] [Help] [answered]
While reading about and trying to implement LL-HLS (for study purposes), I noticed that OME requires both an origin and edge servers to function. I have a couple of questions to satisfy my curiosity:

* Can these edge servers be replaced by NGINX?
* I see there is a protocol (OVT) used to transfer data from origins to edges. Why is this necessary?

I understand how both origin and edge servers work, but I can’t quite grasp the purpose of the edge server.

Another related question: how can segments and manifests be cached effectively?

I’m very familiar with standard HLS and am a maintainer of a [m3u8 Python parser](https://github.com/globocom/m3u8/) for it, but LL-HLS is somewhat new to me.

**Comment by getroot** [comment, marked as answer]
I was just browsing Github, so I can answer this quickly.

You don't need to configure Origin-Edge to use LLHLS. You can use LLHLS streaming with Origin alone, and since it's HTTP-based, you can scale your audience capacity using a Reverse HTTP Proxy Server like Nginx.

OME originally created Edge mode to help scale WebRTC streaming easily (as you know, there is no such thing as a standard WebRTC Cache server.), and the Edge uses self-defined protocol called OVT to replicate the stream from the Origin. Since OVT replicates the stream from the OME Origin to the Edge, the Edge also can provide the LLHLS packaging. However, I always recommend using an HTTP Cache server like Nginx for HTTP-based streaming. It's better in many ways.

**Reply by mauricioabreu** [reply]
Thank you for the detailed answer.
